templates/_helpers.tpl

```diff
@@ -145,16 +145,16 @@
 {{/* --------------- START OF TEMPLATE ------------- */}}
 
 {{- define "gitservers" -}}
-{{- include "joinFmt" (dict "count" (int (default 1 .Values.cluster.gitserver.shards)) "fmt" "gitserver-%d:3178" "sep" " ") -}}
+{{- include "joinFmt" (dict "count" (int (default 1 .Values.cluster.gitserver.shards)) "fmt" "gitserver-%d.gitserver:3178" "sep" " ") -}}
```

Comment:
default should be 0 not 1

Reply:
You should at least have 1 shard though

Reply:
if not specified, don't you want it to print `gitserver-0.gitserver:3178`?

Reply:
It does? See one of the examples: https://github.com/sourcegraph/deploy-sourcegraph/blob/stateful/examples/gcp/generated/sourcegraph/templates/frontend/sourcegraph-frontend.Deployment.yaml#L49-L50

Reply:
I changed `joinFmt` below to count from `[0, n)` instead of `[1, n]`. No other function uses it, so this seemed safe to do. Having the `count` represent a default of a single shard versus the `0`th shard made sense to me